rsc/src/main/java/com/cloudera/livy/rsc/driver/RSCDriver.java

```diff
@@ -172,6 +172,11 @@ public RpcDispatcher onNewClient(Rpc client) {
         registerClient(client);
         return RSCDriver.this;
       }
+
+      @Override
+      public void onSaslComplete(Rpc client) {
```

Comment:
When a new RSCClient connects to RSCDriver, repl should tells server its state. I added this callback to serve this purpose.
It's similar to [onNewClient()](https://github.com/cloudera/livy/pull/231/files/6f72649bfd69abb846f2e93f2e0440278b05829e#diff-fc62b1e9be342e38eeeae8a5f551aa78R271). However, RpcDispatcher isn't set when onNewClient() is called. Rpc.call() can't be called until RpcDispatcher is set. I added the new callback to workaround this issue.
It's called [here].(https://github.com/cloudera/livy/pull/231/files/6f72649bfd69abb846f2e93f2e0440278b05829e#diff-fc62b1e9be342e38eeeae8a5f551aa78R271)
It's called after Sasl authentication is completed.